.github/workflows/pull-request.yml

```diff
@@ -112,4 +112,67 @@ jobs:
         if: env.RUN_JSDOCS == 'True'
         run: echo "Run JSdocs :${{ env.RUN_JSDOCS }}"
 
-          
\ No newline at end of file
+  Dev Server:
+    name: Dev-Server
+    runs-on: ubuntu-latest
+    needs: Test-Application
+    strategy:
+      matrix:
+        node-version: [14.x]
+    services:
+      mongo:
+        image: mongo:4.4
+        options: >-
+          --health-cmd mongo
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 10
+        ports:
+          - 27017:27017
+    env:
+      ACCESS_TOKEN_SECRET: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      REFRESH_TOKEN_SECRET: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      MONGO_DB_URL: "mongodb://localhost:27017/talawa",
+      RECAPTCHA_SITE_KEY: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      RECAPTCHA_SECRET_KEY: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      MAIL_USERNAME: "abc@gmail.com",
+      MAIL_PASSWORD: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      IS_SMTP: "",
+      SMTP_HOST: "",
+      SMTP_PASSWORD: "",
+      SMTP_USERNAME: "",
+      SMTP_PORT: "",
+      SMTP_SSL_TLS: "",
+      apiKey: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      appId: "1:XXXXXXXXXXXX:android:XXXXXXXXXXXXXXXXXXXXXX",
+      messagingSenderId: "XXXXXXXXXXXX",
+      projectId: "appname-XXXXXX",
+      storageBucket: "appname-XXXXXX.appspot.com",
+      iOSapiKey: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+      iOSappId: "1:XXXXXXXXXXXX:ios:XXXXXXXXXXXXXXXXXXXXXX",
+      iOSmessagingSenderId: "XXXXXXXXXXXX",
+      iOSprojectId: "appame-XXXXXX",
+      iOSstorageBucket: "appame-XXXXXX.appspot.com",
+      iosClientId: "XXXXXXXXXXXX-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.apps.googleusercontent.   com",
+      iosBundleId: "com.appname.app",
+      LAST_RESORT_SUPERADMIN_EMAIL: "abc@gmail.com",
+      COLORIZE_LOGS: "true",
+      LOG_LEVEL: "info"
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Sleep for 10s
+        uses: juliangruber/sleep-action@v1
+        with:
+          time: 10s
+
```

Comment:
why we need to wait for 10 seconds, is there anything that will break without it ? @ishtails

Reply:
I don't believe we need to wait, but this sleep command was there in the test job as well before running tests, so I put it here to be on the safe side, as I felt it might be there for some specific reason...

Reply:
I will remove the sleep command from the job then, I also believe it is not required.

That is a good idea, to write a test for it. However I don't have any experience writing tests as of now. Can we have it as a separate issue maybe? So that either someone else could write the test for it, or I could just do it in the future.

Reply:
@ishtails It seems there are some oddities about github actions that I wasn't aware about. The 10 second wait seems to be done for the server to successfully start because the workflow doesn't wait for the server to successfully start I believe? Something like that. 

Also an action just to check if the server successfully starts seems wasteful, it'd be better to have this as a part of some end to end test workflow. For example this action https://github.com/PalisadoesFoundation/talawa-api/blob/develop/.github/workflows/push-documentation.yml already makes use of `npm run dev`. Starting the server itself doesn't tell us anything about whether all the services of the server are working perfectly.

Reply:
I am unaware of any such issues. Let me know if I should put the sleep command there or not...
If someone could explain the need for it, it would be really nice.

The issue originally just stated that there should be a job to check if the dev server starts or not, shouldn't the check passing or failing mean that the server started successfully or failed to start anyway?